lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -202,8 +203,10 @@ class GrandCentralTapsPass : public GrandCentralTapsBase<GrandCentralTapsPass> {
                          InstancePaths &instancePaths);
 
 private:
-  DenseMap<Attribute, BlockArgument> tappedArgs;
-  DenseMap<Attribute, Operation *> tappedOps;
+  using Key = std::pair<Attribute, Attribute>;
```

Comment:
This is an excellent change!

Reply:
Thanks! We could go one step further and just switch to a single `id` space instead of `id` and `portID`. I had originally assumed that there was utility in knowing the `id` of the extmodule that you were tapping _and_ the `portID` within that. However, this was probably poorly thought out and should be changed.

Also, I've tried to use a `pair` or `tuple` as a key before, but I got hung up as `IntegerAttr` _does not have a provided `DenseMap<...>` implementation_. I was very confused until I realized this could just use `Attribute`. 🤷‍♀️